src/plugins/discover/public/application/components/context_app/use_context_app_fetch.tsx

```diff
@@ -0,0 +1,200 @@
+    [config, indexPattern]
+  );
+  const fetchAnchor = useMemo(() => {
+    return fetchAnchorProvider(indexPatterns, searchSource, useNewFieldsApi);
+  }, [indexPatterns, searchSource, useNewFieldsApi]);
+  const { fetchSurroundingDocs } = useMemo(
+    () => fetchContextProvider(indexPatterns, useNewFieldsApi),
+    [indexPatterns, useNewFieldsApi]
+  );
+
+  const [fetchedState, setFetchedState] = useState<ContextFetchState>(
+    getInitialContextQueryState()
+  );
+
+  const setState = useCallback((values: Partial<ContextFetchState>) => {
+    setFetchedState((prevState) => ({ ...prevState, ...values }));
+  }, []);
+
+  const fetchAnchorRow = useCallback(async () => {
+    const { sort } = appState;
+    const [[, sortDir]] = sort;
+    const errorTitle = i18n.translate('discover.context.unableToLoadAnchorDocumentDescription', {
+      defaultMessage: 'Unable to load the anchor document',
+    });
+
+    if (!tieBreakerField) {
+      setState(createError('anchorStatus', FailureReason.INVALID_TIEBREAKER));
+      toastNotifications.addDanger({
+        title: errorTitle,
+        text: toMountPoint(
+          <MarkdownSimple>
+            {i18n.translate('discover.context.invalidTieBreakerFiledSetting', {
+              defaultMessage: 'Invalid tie breaker field setting',
+            })}
+          </MarkdownSimple>
+        ),
+      });
+      return;
+    }
+
+    try {
+      setState({ anchorStatus: { value: LoadingStatus.LOADING } });
+      const anchor = await fetchAnchor(indexPatternId, anchorId, [
+        fromPairs(sort),
+        { [tieBreakerField]: sortDir },
+      ]);
+      setState({ anchor, anchorStatus: { value: LoadingStatus.LOADED } });
+      return anchor;
+    } catch (error) {
+      setState(createError('anchorStatus', FailureReason.UNKNOWN, error));
+      toastNotifications.addDanger({
+        title: errorTitle,
+        text: toMountPoint(<MarkdownSimple>{error.message}</MarkdownSimple>),
+      });
+    }
+  }, [
+    appState,
+    tieBreakerField,
+    setState,
+    toastNotifications,
+    fetchAnchor,
+    indexPatternId,
+    anchorId,
+  ]);
+
+  const fetchSurroundingRows = useCallback(
+    async (type: SurrDocType, fetchedAnchor?: EsHitRecord) => {
+      const filters = filterManager.getFilters();
+      const { sort } = appState;
+      const [[sortField, sortDir]] = sort;
+
+      const count = type === 'predecessors' ? appState.predecessorCount : appState.successorCount;
+      const anchor = fetchedAnchor || fetchedState.anchor;
+      const statusKey = `${type}Status`;
+      const errorTitle = i18n.translate('discover.context.unableToLoadDocumentDescription', {
+        defaultMessage: 'Unable to load documents',
+      });
+
+      if (!tieBreakerField) {
+        setState(createError(statusKey, FailureReason.INVALID_TIEBREAKER));
+        toastNotifications.addDanger({
+          title: errorTitle,
```

Comment:
Do we need to keep `tieBreakerField` check for `fetchSurroundingRows()`? Suppose, we can leave it only for anchor request due to:
- Context rows requests (predecessors and successors) will be fired only after anchor document successfully retrived. It means that `fetchAnchorRow()` will handle empty `tieBreakerField` and send error feedback
- In case when we empty `context:tieBreakerFields` via kibana settings, we need to refresh the page to apply changed settings. So anyway anchor request will be fired first 

On the other hand we can keep it is for stable enhancements in the future (in case when context rows will be queried before anchor doc)